analysisCode/src/EventLoop.cpp

```diff
@@ -46,21 +47,25 @@ int main(int argc, char **argv)
 
       mctree->GetEntry(event);
 
+      TruthEvent trueEvent(*truthEvent);
+      trueEvent.setVerbosity(0);
+      trueEvent.processEvent();
+
       SmearedEvent smearedEvent(*truthEvent, *smearEvent);
       smearedEvent.setVerbosity(0);
       smearedEvent.processEvent();
 
-      
+      PseudoJetVec fjtruthR1Jets = trueEvent.getTruthJets(cs, R1jetdef);
```

Comment:
Lets use the instance of `truthcs` here, instead of `cs`. The `fastjet::ClusterSequence` object contains the jet substructure information, so there should be separate instances of this I believe. We may be able to come up with a better design for using this, also, so that we don't have a bunch of `ClusterSequences` floating around.